src/tests/ftest/control/config_generate_output.py

```diff
@@ -0,0 +1,447 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+'''
+from collections import defaultdict
+
+from apricot import TestWithServers
+from command_utils import CommandFailure
+
+
+class ConfigGenerateOutput(TestWithServers):
+    """Test ID: DAOS-7274
+
+    Verify dmg config generate output.
+
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        """Initialize a ConfigGenerateOutput object."""
+        super().__init__(*args, **kwargs)
+        # Data structure that store expected values.
+        self.nvme_socket_to_addrs = defaultdict(list)
+        self.scm_socket_to_namespaces = defaultdict(list)
+        self.pci_address_set = set()
+        self.scm_namespace_set = set()
+
+        self.interface_to_providers = defaultdict(list)
+        self.interface_set = set()
+
+    def prepare_expected_data(self):
+        """Prepare expected values.
+
+        Call dmg storage scan and network scan to collect expected values.
+        """
+        dmg = self.get_dmg_command()
+
+        # Call dmg storage scan --json.
+        storage_out = dmg.storage_scan()
+
+        # Get nvme_devices and scm_namespaces list that are buried. There's a
+        # uint64 hash of the strcut under HostStorage.
+        temp_dict = storage_out["response"]["HostStorage"]
+        struct_hash = list(temp_dict.keys())[0]
+        nvme_devices = temp_dict[struct_hash]["storage"]["nvme_devices"]
+        scm_namespaces = temp_dict[struct_hash]["storage"]["scm_namespaces"]
+
+        # Fill in the dictionary and the set for NVMe.
+        for nvme_device in nvme_devices:
+            socket_id = nvme_device["socket_id"]
+            pci_addr = nvme_device["pci_addr"]
+            self.nvme_socket_to_addrs[socket_id].append(pci_addr)
+            self.pci_address_set.add(pci_addr)
+
+        self.log.debug(
+            "nvme_socket_to_addrs = {}".format(self.nvme_socket_to_addrs))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions